servers/src/main/java/tachyon/master/TachyonMaster.java

```diff
@@ -102,23 +102,24 @@ public TachyonMaster(TachyonConf tachyonConf) {
     TachyonConf.assertValidPort(port, mTachyonConf);
     TachyonConf.assertValidPort(webPort, mTachyonConf);
 
-    String hostnameExternal = mTachyonConf.get(Constants.MASTER_HOSTNAME, "localhost");
-    // The master listens to the MASTER_HOSTNAME_LISTENING configuration option.
-    // MASTER_HOSTNAME_LISTENING defaults to MASTER_HOSTNAME if unspecified. If set to
-    // MASTER_HOSTNAME_LISTENING_WILDCARD, server listens to all addresses.
-    String hostnameListening =
-        mTachyonConf.get(Constants.MASTER_HOSTNAME_LISTENING, hostnameExternal);
-
-    InetSocketAddress address = new InetSocketAddress(hostnameExternal, port);
-    InetSocketAddress addressListening;
-    if (hostnameListening.equals(Constants.MASTER_HOSTNAME_LISTENING_WILDCARD)) {
-      addressListening = new InetSocketAddress(port);
-    } else {
-      addressListening = new InetSocketAddress(hostnameListening, port);
-    }
+    // String hostnameExternal = mTachyonConf.get(Constants.MASTER_HOSTNAME, "localhost");
+    // // The master listens to the MASTER_HOSTNAME_LISTENING configuration option.
+    // // MASTER_HOSTNAME_LISTENING defaults to MASTER_HOSTNAME if unspecified. If set to
+    // // MASTER_HOSTNAME_LISTENING_WILDCARD, server listens to all addresses.
+    // String hostnameListening =
+    // mTachyonConf.get(Constants.MASTER_HOSTNAME_LISTENING, hostnameExternal);
+    //
+    // InetSocketAddress address = new InetSocketAddress(hostnameExternal, port);
+    // InetSocketAddress addressListening;
+    // if (hostnameListening.equals(Constants.MASTER_HOSTNAME_LISTENING_WILDCARD)) {
+    // addressListening = new InetSocketAddress(port);
+    // } else {
+    // addressListening = new InetSocketAddress(hostnameListening, port);
+    // }
+    InetSocketAddress masterBindAddress = NetworkAddressUtils.getMasterBindAddress(mTachyonConf);
```

Comment:
It looks that I don't really catch your question. Could you give an example?  This modification should not change original logic.